Add new community guide for using RgGen

Cra2yPierr0t opened the pull request:
This is new community guide.

vijayank88 commented:
@Cra2yPierr0t 
Can you do one more update here https://vlsi.jp/Introduction_to_OpenMPW.html#installing-caravel
Give a note to select respective branch for clone caravel user project for sky130/gf180 to avoid mis configuration or precheck issues. One of the user using your tutorial and facing issue for gf180 submission.
```
git clone https://github.com/efabless/caravel_user_project.git
```
update to
```
git clone -branch <tag> https://github.com/efabless/caravel_user_project.git
```
Select latest tag from here https://github.com/efabless/caravel_user_project/tags based on MPW shuttle like for GFMPW-0 its `gfmpw-0d` and for SKY130 MPW-8 its `mpw-8c`

So for each shuttle the respective tag will be updated and user should adopt the tags to avoid precheck errors

Cra2yPierr0t commented:
Thanks for the comment.
In addition to clone branch, fixed broken links and added description of `user_define.v`.
There are still some not-so-good descriptions left, so I will update them as I go along.

donn reviewed (approved):
Thank you so much!
